junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/EnumArgumentsProvider.java

```diff
@@ -43,7 +43,13 @@ protected Stream<? extends Arguments> provideArguments(ExtensionContext context,
 
 	private <E extends Enum<E>> Set<? extends E> getEnumConstants(ExtensionContext context, EnumSource enumSource) {
 		Class<E> enumClass = determineEnumClass(context, enumSource);
-		return EnumSet.allOf(enumClass);
+		E[] constants = enumClass.getEnumConstants();
+		if (constants.length == 0) {
+			return EnumSet.noneOf(enumClass);
```

Comment:
We should validate that both `from` and `to` are empty in this case, shouldn't we?